csclient/csclient.go

```diff
@@ -156,6 +158,126 @@ func (c *Client) GetArchive(id *charm.URL) (r io.ReadCloser, eid *charm.URL, has
 	return resp.Body, eid, hash, resp.ContentLength, nil
 }
 
+// ListResources retrieves the metadata about resources for the given charm.
+func (c *Client) ListResources(ids ...*charm.URL) (map[string][]params.Resource, error) {
```

Comment:
Why the ...?  We aren't supporting "give me all charms".  And there isn't any precedent here, is there?  (charmrepo.CharmStore.Latest doesn't count.)